synapse/third_party_rules/access_rules.py

```diff
@@ -170,12 +196,23 @@ def on_create_room(self, requester, config, is_requester_admin) -> bool:
         return True
 
     @defer.inlineCallbacks
-    def check_threepid_can_be_invited(self, medium, address, state_events):
-        """Implements synapse.events.ThirdPartyEventRules.check_threepid_can_be_invited
+    def check_threepid_can_be_invited(
+        self, medium: str, address: str, state_events: Dict[Tuple[str, str], EventBase],
```

Comment:
There's a StateMap type here, which is helpful to show the intention:
```suggestion
        self, medium: str, address: str, state_events: StateMap[EventBase],
```

Reply:
(likewise below)

Reply:
Ah, that's much cleaner, thank you!